### PR TITLE
falco app installation

### LINKED ISSUE #347: [feature request] we need to add falco as an app
we can support Falco the container runtime security tool as an app, they have charts that we can use.
https://github.com/falcosecurity/charts 

## Expected Behaviour

```bash
$ arkade install falco --set falcosidekick.enabled=true
```
This command should install the Falco to the cluster using its Helm chart.

## Current Behaviour

There is no solution exist yet for Falco.

## Possible Solution

We can use the Helm charts of Falco to install it to the cluster. I mentioned the link above for the Helm charts of the Falco.

## Context

Falco is a very popular project in the container runtime security area, so, it would be nice if we can support the installation of the Falco in arkade.

## Your Environment

* What Kubernetes distribution are you using?

```
$ kubectl version
Client Version: version.Info{Major:"1", Minor:"20", GitVersion:"v1.20.4", GitCommit:"e87da0bd6e03ec3fea7933c4b5263d151aafd07c", GitTreeState:"clean", BuildDate:"2021-02-18T16:12:00Z", GoVersion:"go1.15.8", Compiler:"gc", Platform:"darwin/amd64"}
Server Version: version.Info{Major:"1", Minor:"20", GitVersion:"v1.20.2+k3s1", GitCommit:"1d4adb0301b9a63ceec8cabb11b309e061f43d5f", GitTreeState:"clean", BuildDate:"2021-01-14T23:52:37Z", GoVersion:"go1.15.5", Compiler:"gc", Platform:"linux/amd64"}
```

* Operating System and version (e.g. Linux, Windows, MacOS):

```
$ uname -a
Darwin Batuhan-MBP.home 19.6.0 Darwin Kernel Version 19.6.0: Tue Jan 12 22:13:05 PST 2021; root:xnu-6153.141.16~1/RELEASE_X86_64 x86_64
```

### PR DESCRIPTION
**Pull request by developer-guy**
falco app installation
## Motivation and Context

fixes #347 
- [x] I have raised an issue to propose this change ([required](https://github.com/openfaas/faas/blob/master/CONTRIBUTING.md))

## How Has This Been Tested?

first, build an app, then run the command.

```bash
$ civo kubernetes create playground --size=g3.k3s.medium --nodes=1 --region NYC1 --wait
$ civo kubernetes config playground --save --merge
$ go build -o ark
$ ./ark install falco --sidekick=true --webui=true
$ kubectl get pods
```
Screen Shot 2021-03-16 at 23 14 24
Screen Shot 2021-03-16 at 23 14 57

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I've read the [CONTRIBUTION](https://github.com/openfaas/faas/blob/master/CONTRIBUTING.md) guide
- [x] I have signed-off my commits with `git commit -s`
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

- [] I have tested this on arm, or have added code to prevent deployment

### CODE REVIEW

**Comment by danpopSD** [comment]
bravo!

**Comment by alexellis** [comment]
@developer-guy thanks for suggesting this app. Can you please fill out the whole template?

I've asked @Waterdrips to do a review, which is required before a merge.

**Comment by alexellis** [comment]
See the comment on the issue please - https://github.com/alexellis/arkade/issues/347

**Comment by developer-guy** [comment]
@alexellis @Waterdrips done.

**Review by nitishkumar71 — APPROVED** [review]
LGTM

File: cmd/apps/falco.go [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+
+		fmt.Println(falcoInstallMsg)
+		return nil
+	}
+
+	return falco
+}
+
+var FalcoInfoMsg = `
+[Falco](https://falco.org) is a *Cloud Native Runtime Security* tool designed to detect anomalous activity in your applications.
```
Comment (alexellis) [inline cmd/apps/falco.go:90]: We can't  show markdown links in a fmt.Println, can you remove it please?

File: cmd/apps/falco.go [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+[Falco](https://falco.org) is a *Cloud Native Runtime Security* tool designed to detect anomalous activity in your applications. 
+You can use Falco to monitor runtime security of your Kubernetes applications and internal components.
+
+# Find out more at:
+https://github.com/falcosecurity/falco
+
+# Learn about how to use Falco
+https://falco.org/docs/
+
+# Uninstall Falco
```
Comment (alexellis) [inline cmd/apps/falco.go:99]: This repetition is unnecessary, can you remove it?

**Comment by alexellis** [comment]
Remapping complex values behind simple flags is part of the value prop of arkade, and we should keep consistency here.

Can you add the suggested flag please and I'll go ahead and merge.

> You mentioned -set falcosidekick.enabled=true - but this is an example of what the flags are for, to hide away the nested --set commands, so this would be: --sidekick as in arkade install falco --sidekick=true/false

**Review by alexellis — CHANGES_REQUESTED** [review]
Please see comments.

File: cmd/apps/falco.go [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+
+	falco.RunE = func(command *cobra.Command, args []string) error {
+		kubeConfigPath, _ := command.Flags().GetString("kubeconfig")
+		wait, _ := command.Flags().GetBool("wait")
+		updateRepo, _ := command.Flags().GetBool("update-repo")
+		customFlags, _ := command.Flags().GetStringArray("set")
+
+		fmt.Printf("Using kubeconfig: %s\n", kubeConfigPath)
+		arch := k8s.GetNodeArchitecture()
+		fmt.Printf("Node architecture: %q\n", arch)
```
Comment (alexellis) [inline cmd/apps/falco.go:62]: Can the Falco chart and sidekick run on Raspberry Pi and ARM64? If not, we need to block that here - see the codebase for examples of how.

File: cmd/apps/falco.go (hunk 2 above)
Comment (developer-guy) [reply cmd/apps/falco.go]: done.

**Comment by developer-guy** [comment]
> Remapping complex values behind simple flags is part of the value prop of arkade, and we should keep consistency here.
> 
> Can you add the suggested flag please and I'll go ahead and merge.
> 
> > You mentioned -set falcosidekick.enabled=true - but this is an example of what the flags are for, to hide away the nested --set commands, so this would be: --sidekick as in arkade install falco --sidekick=true/false

I added this 👍

File: cmd/apps/falco.go (hunk 1 above)
Comment (developer-guy) [reply cmd/apps/falco.go]: done

File: cmd/apps/falco.go (hunk 3 above)
Comment (developer-guy) [reply cmd/apps/falco.go:64]: done, they can not run on pi and arm64 now.

**Review by alexellis — APPROVED** [review]
Approved

**Pull request merged**